packages/config/src/api/site_info.ts

```diff
@@ -167,15 +167,21 @@ const getIntegrations = async function ({
     ? `${baseUrl}team/${accountId}/integrations/installations/meta/${siteId}`
     : `${baseUrl}site/${siteId}/integrations/safe`
 
+  let response
   try {
-    const response = await fetch(url)
+    response = await fetch(url)
+    if (response.status !== 200) {
+      throw new Error(`Unexpected status code ${response.status} from fetching extensions`)
+    }
+  } catch (error) {
+    throwUserError(`Failed retrieving extensions for site ${siteId}: ${error.message}. ${ERROR_CALL_TO_ACTION}`)
+  }
 
-    const integrations = await response.json()
-    return Array.isArray(integrations) ? integrations : []
+  try {
+    if (Number(response.headers.get(`content-length`)) === 0) return []
+    const responseBody = await response.json()
+    return Array.isArray(responseBody) ? responseBody : []
   } catch (error) {
-    // Integrations should not block the build if they fail to load
-    // TODO: We should consider blocking the build as integrations are a critical part of the build process
-    // https://linear.app/netlify/issue/CT-1214/implement-strategy-in-builds-to-deal-with-integrations-that-we-fail-to
-    return []
+    throwUserError(`Failed to parse extensions for site ${siteId}: ${error.message}. ${ERROR_CALL_TO_ACTION}`)
```

Comment:
Learning - What is the failure scenario here? Response is a 200, but the integration returns an object that is not parsable to json? How can this occur?

Since it's potentially expected behaviour, could we add a test case for this scenario?

Reply:
This is what we return from the (https://github.com/netlify/jigsaw/blob/main/api/team_installations.go#L469-L559, and it seems that if we return a 200 status, it should always return an object that is parseable as JSON.

I was a bit wary of how it was written before, where we caught all errors and returned an empty array. This made me wonder if there were some edge cases I wasn't aware of when I added this.

I'll add some tests to cover this scenario.

Reply:
Since this is kind of a nit-like ask - feel free to do this work in a follow up PR